handler/handler.go

```diff
@@ -108,3 +113,65 @@ func writeJSON(status int, data interface{}, w http.ResponseWriter) error {
 	w.WriteHeader(status)
 	return json.NewEncoder(w).Encode(data)
 }
+
+func renderHTML(w http.ResponseWriter, r *http.Request, task eremeticTask, taskID string) {
+	var err error
+	var tpl *template.Template
+
+	data := make(map[string]interface{})
+	funcMap := template.FuncMap{
+		"Label": LabelColor,
+	}
+
+	if task == (eremeticTask{}) {
+		tpl, err = template.ParseFiles("templates/error_404.html")
+		data["TaskID"] = taskID
+	} else {
+		tpl, err = template.New("task.html").Funcs(funcMap).ParseFiles("templates/task.html")
+		data = makeMap(task)
+	}
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Error(err.Error())
+		return
+	}
+
+	err = tpl.Execute(w, data)
+}
+
+func makeMap(task eremeticTask) map[string]interface{} {
+	data := make(map[string]interface{})
+
+	data["TaskID"] = task.ID
+	data["CommandEnv"] = task.Command.GetEnvironment().GetVariables()
+	data["CommandUser"] = task.Command.GetUser()
+	data["Command"] = task.Command.GetValue()
+	// TODO: Support more than docker?
+	data["ContainerImage"] = task.Container.GetDocker().GetImage()
+	data["FrameworkID"] = task.FrameworkId
+	data["Hostname"] = task.Hostname
+	data["Name"] = task.Name
+	data["SlaveID"] = task.SlaveId
+	data["Status"] = task.Status
+	data["CPU"] = fmt.Sprintf("%.2f", task.TaskCPUs)
+	data["Memory"] = fmt.Sprintf("%.2f", task.TaskMem)
+
+	return data
+}
+
+// LabelColor is used to map a status to a label color
+func LabelColor(status string) string {
+	switch status {
+	case "TASK_FAILED":
+		return "red"
```

Comment:
colours feels more like a css thing. should the class perhaps just be "task-failed" or so?